File: 10235/10235.pas
var
	a:array[0..1000000] of longint;
	N:longint;

procedure init();
var
	i,j:longint;
begin
	for i:=0 to 1000000 do a[i]:=1;
	for i:=2 to 1000000 do begin
		if a[i]=0 then continue;
		j:=2;
		while i*j<1000000 do begin
			a[i*j]:=0;
			j:=j+1;
		end;
	end;
end;

function reverse(v:longint):longint;
var
	t:longint;
begin
	t:=0;
	while v<>0 do begin
		t:=t*10+v mod 10;
		v:=v div 10;
	end;
	exit(t);
end;

function prime(n:longint):longint;
begin
	if a[n]=0 then exit(0)
	else if n=reverse(n) then exit(1)
	else if a[reverse(n)]=0 then exit(1)
	else exit(2);
end;

begin
	init();
	while not eof do begin
		readln(N);
		case prime(N) of 
			0:writeln(N,' is not prime.');
			1:writeln(N,' is prime.');
			2:writeln(N,' is emirp.');
		end;
	end;
end.

File: 10235/10235.cpp
#include<iostream>
using namespace std;

int a[1000000];
int N;
void init()
{
	for(int i=0;i<1000000;i++)a[i]=1;
	for(long long i=2;i<1000000;i++){
		if(a[i]==0)continue;
		for(long long j=2;i*j<1000000;j++)a[i*j]=0;
	}
}
int reverse(int v)
{
	int t=0;
	while(v){
		t=t*10+v%10;
		v/=10;
	}
	return t;
}
int prime(int n)
{
	if(a[n]==0)return 0;
	else if(n==reverse(n))return 1;
	else if(a[reverse(n)]==0)return 1;
	else return 2;	
}
int main()
{
	init();
	while(cin>>N){
		switch(prime(N)){
			case 0:cout<<N<<" is not prime.\n";
			       break;
			case 1:cout<<N<<" is prime.\n";
			       break;
			case 2:cout<<N<<" is emirp.\n";
			       break;
			default:cout<<"error\n";
		}
	}
	return 0;
}
